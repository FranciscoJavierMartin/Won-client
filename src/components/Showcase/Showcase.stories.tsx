import { ComponentMeta, Story } from '@storybook/react';
import Showcase, { ShowcaseProps } from '.';
import highlightMock from '@/components/Highlight/mock';
import gamesMock from '@/components/GameCardSlider/mock';

export default {
  title: 'Showcase',
  component: Showcase,
  args: {},
  decorators: [
    (Story) => (
      <div style={{ margin: '0 auto' }}>
        <Story />
      </div>
    ),
  ],
  parameters: {
    layout: 'fullscreen',
    backgrounds: {
      default: 'won-dark',
    },
  },
} as ComponentMeta<typeof Showcase>;

export const Default: Story<ShowcaseProps> = (args) => <Showcase {...args} />;

Default.args = {
  title: 'Most popular',
  highlight: highlightMock,
  games: gamesMock,
};
