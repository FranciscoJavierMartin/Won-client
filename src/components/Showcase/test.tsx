import { screen } from '@testing-library/react';
import Showcase from '.';
import highlightMock from '@/components/Highlight/mock';
import gamesMock from '@/components/GameCardSlider/mock';
import { renderWithTheme } from '@/utils/tests/helpers';

const props = {
  title: 'Most popular',
  highlight: highlightMock,
  games: gamesMock.slice(0, 1),
};

describe('<Showcase />', () => {
  it('should render full showcase', () => {
    const { container } = renderWithTheme(<Showcase {...props} />);

    expect(
      screen.getByRole('heading', { name: /most popular/i })
    ).toBeInTheDocument();
    expect(
      screen.getByRole('heading', { name: highlightMock.title })
    ).toBeInTheDocument();
    expect(
      screen.getByRole('heading', { name: gamesMock[0].title })
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render without highlight', () => {
    renderWithTheme(<Showcase title={props.title} games={props.games} />);
    expect(
      screen.getByRole('heading', { name: /most popular/i })
    ).toBeInTheDocument();
    expect(
      screen.queryByRole('heading', { name: highlightMock.title })
    ).not.toBeInTheDocument();
    expect(
      screen.getByRole('heading', { name: gamesMock[0].title })
    ).toBeInTheDocument();
  });

  it('should render without games', () => {
    renderWithTheme(
      <Showcase title={props.title} highlight={props.highlight} />
    );
    expect(
      screen.getByRole('heading', { name: /most popular/i })
    ).toBeInTheDocument();
    expect(
      screen.getByRole('heading', { name: highlightMock.title })
    ).toBeInTheDocument();
    expect(
      screen.queryByRole('heading', { name: gamesMock[0].title })
    ).not.toBeInTheDocument();
  });
});
