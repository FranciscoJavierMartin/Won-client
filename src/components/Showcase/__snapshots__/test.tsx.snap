// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Showcase /> should render the heading 1`] = `
<div
  class=""
>
  <h1>
    Showcase
  </h1>
</div>
`;
